contracts/FlowIDTableStaking.cdc

```diff
@@ -1069,15 +1071,17 @@ pub contract FlowIDTableStaking {
             
             let allNodeIDs = FlowIDTableStaking.getNodeIDs()
 
+            let stakedNodeIDs: {String: Bool} = FlowIDTableStaking.getParticipantNodeList()!
+
             for nodeID in allNodeIDs {
                 let nodeRecord = FlowIDTableStaking.borrowNodeRecord(nodeID)
 
-                FlowIDTableStaking.totalTokensStakedByNodeType[nodeRecord.role] = FlowIDTableStaking.totalTokensStakedByNodeType[nodeRecord.role]! + nodeRecord.tokensCommitted.balance
-
                 // mark the committed tokens as staked
                 if nodeRecord.tokensCommitted.balance > 0.0 {
+                    FlowIDTableStaking.totalTokensStakedByNodeType[nodeRecord.role] = FlowIDTableStaking.totalTokensStakedByNodeType[nodeRecord.role]! + nodeRecord.tokensCommitted.balance
                     emit TokensStaked(nodeID: nodeRecord.id, amount: nodeRecord.tokensCommitted.balance)
                     nodeRecord.tokensStaked.deposit(from: <-nodeRecord.tokensCommitted.withdraw(amount: nodeRecord.tokensCommitted.balance))
```

Comment:
Instead of moving the token from one vault to another, why not keep the token in one vault and just update the variable? In that way, it is cheaper and faster.

Reply:
We originally did it this way because we wanted to have the protections that resources have for managing these tokens. I agree that it is cheaper and faster to do it with your suggestion, but this is what we're stuck with unless we do a big refactor

Reply:
Cool, Make sense